pkg/hubble/server/serveroption/option.go

```diff
@@ -55,6 +56,8 @@ type Options struct {
 	OnMonitorEvent []OnMonitorEvent        // invoked before an event is decoded
 	OnDecodedFlow  []OnDecodedFlow         // invoked after a flow has been decoded
 	OnBuildFilter  []filters.OnBuildFilter // invoked while building a flow filter
+	OnFlowDelivery []OnDecodedFlow         // invoked before a flow is delivered via API
```

Comment:
nit: I find it a bit weird to have the same interface for two semantically very different hooks. For example, a `OnDecodedFlow` hook is allowed to modify the flow (since at that point it is not yet part of the ringbuffer), while a `OnFlowDelivery` must never write to the flow.

I would prefer it if we had a separate interface for `OnFlowDelivery`, even if it has the same exact signature as `OnDecodedFlow`.